backend/clickhouse/migrations/000136_add_metrics_aggregate_mv.up.sql

```diff
@@ -0,0 +1,107 @@
+CREATE TABLE IF NOT EXISTS default.metrics
+(
+    `ProjectId`         UInt32,
+    `ServiceName`       String,
+    `MetricName`        String,
+    `MetricType`        Enum8('Empty' = 0, 'Gauge' = 1, 'Sum' = 2, 'Histogram' = 3, 'ExponentialHistogram' = 4, 'Summary' = 5),
+    `Attributes`        Map(LowCardinality(String), String),
+    `Timestamp`         DateTime CODEC (Delta(4), ZSTD(1)),
+    -- meta
+    `MetricDescription` SimpleAggregateFunction(anyLast, String),
+    `MetricUnit`        SimpleAggregateFunction(anyLast, String),
+    `StartTimestamp`    SimpleAggregateFunction(min, DateTime64(9)) CODEC (Delta(8), ZSTD(1)),
+    `RetentionDays`     SimpleAggregateFunction(max, UInt8) DEFAULT 30,
+    `Exemplars.Attributes` SimpleAggregateFunction(groupArrayArray, Array(Map(String, String))),
+    `Exemplars.Timestamp` SimpleAggregateFunction(groupArrayArray, Array(DateTime64(9))),
+    `Exemplars.Value` SimpleAggregateFunction(groupArrayArray, Array(Float64)),
+    `Exemplars.SpanID` SimpleAggregateFunction(groupArrayArray, Array(String)),
+    `Exemplars.TraceID` SimpleAggregateFunction(groupArrayArray, Array(String)),
+    `Exemplars.SecureSessionID` SimpleAggregateFunction(groupArrayArray, Array(String)),
+    -- histogram
+    `Min`               SimpleAggregateFunction(min, Float64),
+    `Max`               SimpleAggregateFunction(max, Float64),
+    `BucketCounts`      SimpleAggregateFunction(groupArrayArray, Array(UInt64)),
+    `ExplicitBounds`    SimpleAggregateFunction(groupArrayArray, Array(Float64)),
+    -- summary
+    `ValueAtQuantiles.Quantile` SimpleAggregateFunction(groupArrayArray, Array(Float64)),
+    `ValueAtQuantiles.Value` SimpleAggregateFunction(groupArrayArray, Array(Float64)),
+    -- common
+    `Count`             SimpleAggregateFunction(sum, UInt64),
+    `Sum`               SimpleAggregateFunction(sum, Float64)
+)
+    ENGINE = AggregatingMergeTree()
+    PARTITION BY toStartOfDay(Timestamp)
+    ORDER BY (ProjectId, ServiceName, MetricName, MetricType, Attributes, toUnixTimestamp(Timestamp))
+    TTL toDateTime(Timestamp) + toIntervalDay(RetentionDays);
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS metrics_sum_mv TO metrics AS
+SELECT ProjectId,
+       ServiceName,
+       MetricName,
+       MetricType,
+       Attributes,
+       toDateTime(toStartOfSecond(Timestamp))                as Timestamp,
+       -- meta
+       anyLastSimpleState(MetricDescription)                 as MetricDescription,
```

Comment:
what's the difference between `any` and `anyLast` here?

Reply:
looks like both `any` and `anyLast` are supported via [SimpleAggregateFunction](https://clickhouse.com/docs/en/sql-reference/data-types/simpleaggregatefunction)

[any](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/any#agg_function-any) returns the first row
[anyLast](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/anylast#anylastx) returns the last row

tho looks like both are technically non-deterministic?